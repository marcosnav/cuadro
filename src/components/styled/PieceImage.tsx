import styled, { StyledProps } from 'styled-components';

interface IProps {
  left: number;
  top: number;
  image: string;
}

const PieceImage = styled.div`
  background: ${(props) => props.theme.DARK};
  background-image: url(${({ image }: StyledProps<IProps>) => image});
  background-position: center center;
  background-size: cover cover;
  height: 268px;
  position: relative;
  transform: ${({top, left}: StyledProps<IProps>) => `translate(${left}px,${top}px);`}
  width: 268px;
`;

PieceImage.displayName = 'PieceImage';

export default PieceImage;
