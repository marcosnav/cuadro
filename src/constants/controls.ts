/**
 * Available control types
 */
export enum ControlType {
  NEW_GAME = 'NEW_GAME',
  RESTART_GAME = 'RESTART_GAME',
  SEE_ORIGINAL = 'SEE_ORIGINAL',
}
