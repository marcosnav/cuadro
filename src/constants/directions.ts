/**
 * SwipeDirection, one of UP, RIGHT, LEFT, DOWN
 */
export enum SwipeDirection {
  UP = 'UP',
  RIGHT = 'RIGHT',
  DOWN = 'DOWN',
  LEFT = 'LEFT',
}
