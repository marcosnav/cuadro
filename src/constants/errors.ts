export enum errors {
  NO_INTERNET = 'NO_INTERNET',
  UNEXPECTED_ERROR = 'UNEXPECTED_ERROR',
  UNRESPONSIVE_IMAGE_PROVIDER = 'UNRESPONSIVE_IMAGE_PROVIDER',
}
