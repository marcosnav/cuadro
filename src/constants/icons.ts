import check from './../assets/svg/check.svg';
import image from './../assets/svg/image.svg';
import newicon from './../assets/svg/new.svg';
import refresh from './../assets/svg/refresh.svg';

export const icons: { [key: string]: string } = {
  check,
  image,
  new: newicon,
  refresh,
};
