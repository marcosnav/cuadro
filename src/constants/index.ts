export * from './status';
export * from './errors';
export * from './theme';
export * from './keycodes';
