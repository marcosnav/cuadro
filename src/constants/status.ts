export enum status {
  LOADING_IMAGE = 'LOADING_IMAGE',
  DISPLAY_ORIGINAL = 'DISPLAY_ORIGINAL',
  PLAYING_GAME  = 'PLAYING_GAME ',
  FINISHED_GAME = 'FINISHED_GAME',
}
