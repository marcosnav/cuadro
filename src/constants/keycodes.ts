import { SwipeDirection } from './directions';

/**
 * KeyCodes to match for keypress events
 */
export enum KeyCodes {
  DOWN = 40,
  LEFT = 37,
  RIGHT = 39,
  UP = 38,
  ENTER = 13,
}

/**
 * Map key code with swipe direction
 */
export const KeySwipeMap: { [key: number]: SwipeDirection } = {
  40: SwipeDirection.DOWN,
  37: SwipeDirection.LEFT,
  39: SwipeDirection.RIGHT,
  38: SwipeDirection.UP,
};
