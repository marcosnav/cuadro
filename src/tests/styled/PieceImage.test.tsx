import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import { ThemeConsumer } from 'styled-components';
import PieceImage from './../../components/styled/PieceImage';
import { theme } from './../../constants';

describe('<PieceImage/>', () => {
  beforeAll(() => {
    ThemeConsumer._currentValue = theme;
  });

  test('renders properly', () => {
    const component = mount(<PieceImage image={'cat-picture.jpg'} top={0} left={201} />);
    expect(component).toHaveStyleRule('background', theme.DARK);
    expect(component).toHaveStyleRule('background-image', 'url(cat-picture.jpg)');
    expect(component).toHaveStyleRule('transform', 'translate(201px,0px)');
    expect(component).toHaveStyleRule('height', '268px');
    expect(component).toHaveStyleRule('width', '268px');
    expect(component).toMatchSnapshot();
  });
});
