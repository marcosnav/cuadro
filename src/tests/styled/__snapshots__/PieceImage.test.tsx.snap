// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PieceImage/> renders properly 1`] = `
.c0 {
  background: #303030;
  background-image: url(cat-picture.jpg);
  background-position: center center;
  background-size: cover cover;
  height: 268px;
  position: relative;
  -webkit-transform: translate(201px,0px);
  -ms-transform: translate(201px,0px);
  transform: translate(201px,0px);
  width: 268px;
}

<PieceImage
  image="cat-picture.jpg"
  left={201}
  top={0}
>
  <div
    className="c0"
  />
</PieceImage>
`;
